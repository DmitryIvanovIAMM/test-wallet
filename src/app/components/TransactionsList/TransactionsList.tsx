import { Transaction } from '@/types/types';
import TransactionSmallCard from '@/app/components/Transaction/TransactionSmallCard';

export interface TransactionsListProps {
  transactions: Transaction[];
}

export default function TransactionsList({ transactions }: TransactionsListProps) {
  console.log('TransactionsList().  transactions:', transactions);

  return (
    <div className="grid grid-cols-1 gap-4 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4">
      {transactions.map((transaction: Transaction) => (
        <TransactionSmallCard key={transaction._id} transaction={transaction} />
      ))}
    </div>
  );
}
