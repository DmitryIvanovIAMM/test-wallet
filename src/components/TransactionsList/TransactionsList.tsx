import { Transaction, User } from '@/types/types';
import TransactionSmallCard from '@/components/TransactionSmallCard/TransactionSmallCard';

export interface TransactionsListProps {
  user: User;
  transactions: Transaction[];
}

export default function TransactionsList({ user, transactions }: TransactionsListProps) {
  return (
    <>
      <div
        className="grid grid-cols-1 gap-0 sm:grid-cols-2 align-text-left font-bold text-gray-800 text-2xl"
        data-testid="transactions-list-header"
      >
        Latest transactions
      </div>
      <div
        className="grid grid-cols-1 gap-0 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 rounded-md"
        data-testid="transactions-list"
      >
        {transactions.map((transaction: Transaction, i: number) => (
          <TransactionSmallCard
            key={`${i}-transaction-${transaction._id}`}
            user={user}
            transaction={transaction}
          />
        ))}
      </div>
    </>
  );
}
