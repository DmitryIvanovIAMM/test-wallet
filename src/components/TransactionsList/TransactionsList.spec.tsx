import { render } from '@testing-library/react';
import TransactionsList, {
  TransactionsListProps
} from '@/components/TransactionsList/TransactionsList';
import {
  authorizesUser,
  testCreditTransaction,
  testPaymentTransaction
} from '@/test-data/testHelpers';

const setup = (propsOverride: Partial<TransactionsListProps> = {}) => {
  const props: TransactionsListProps = {
    user: authorizesUser,
    transactions: [testPaymentTransaction, testCreditTransaction],
    ...propsOverride
  };

  const container = render(<TransactionsList {...props} />);

  return {
    props,
    container
  };
};

describe('TransactionsList component should', () => {
  it('render correctly', async () => {
    const { container } = await setup();

    expect(container).toMatchSnapshot();
  });

  it('render show data', async () => {
    const { container, props } = await setup();

    expect(container.getByTestId('transactions-list-header')).toBeInTheDocument();
    expect(container.getByText('Latest transactions')).toBeInTheDocument();
    expect(container.getByTestId('transactions-list')).toBeInTheDocument();

    expect(container.getByText(props.transactions[0].transactionDescription)).toBeInTheDocument();
    expect(container.getByText(props.transactions[0].transactionName)).toBeInTheDocument();
    expect(container.getByText(props.transactions[1].transactionDescription)).toBeInTheDocument();
    expect(container.getByText(props.transactions[1].transactionName)).toBeInTheDocument();
  });
});
