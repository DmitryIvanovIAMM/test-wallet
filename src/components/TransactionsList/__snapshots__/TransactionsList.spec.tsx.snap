// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionsList component should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="grid grid-cols-1 gap-0 sm:grid-cols-2 align-text-left font-bold text-gray-800 text-2xl"
        data-testid="transactions-list-header"
      >
        Latest transactions
      </div>
      <div
        class="grid grid-cols-1 gap-0 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 rounded-md"
        data-testid="transactions-list"
      >
        <div
          class="flex-row gap-1 p-1 bg-white grid grid-cols-10"
        >
          <div
            class="col-span-2 gap-1 p-1"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-cc-visa w-full h-full"
              color="black"
              data-icon="cc-visa"
              data-prefix="fab"
              data-testid="transaction-icon"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M470.1 231.3s7.6 37.2 9.3 45H446c3.3-8.9 16-43.5 16-43.5-.2.3 3.3-9.1 5.3-14.9l2.8 13.4zM576 80v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V80c0-26.5 21.5-48 48-48h480c26.5 0 48 21.5 48 48zM152.5 331.2L215.7 176h-42.5l-39.3 106-4.3-21.5-14-71.4c-2.3-9.9-9.4-12.7-18.2-13.1H32.7l-.7 3.1c15.8 4 29.9 9.8 42.2 17.1l35.8 135h42.5zm94.4.2L272.1 176h-40.2l-25.1 155.4h40.1zm139.9-50.8c.2-17.7-10.6-31.2-33.7-42.3-14.1-7.1-22.7-11.9-22.7-19.2.2-6.6 7.3-13.4 23.1-13.4 13.1-.3 22.7 2.8 29.9 5.9l3.6 1.7 5.5-33.6c-7.9-3.1-20.5-6.6-36-6.6-39.7 0-67.6 21.2-67.8 51.4-.3 22.3 20 34.7 35.2 42.2 15.5 7.6 20.8 12.6 20.8 19.3-.2 10.4-12.6 15.2-24.1 15.2-16 0-24.6-2.5-37.7-8.3l-5.3-2.5-5.6 34.9c9.4 4.3 26.8 8.1 44.8 8.3 42.2.1 69.7-20.8 70-53zM528 331.4L495.6 176h-31.1c-9.6 0-16.9 2.8-21 12.9l-59.7 142.5H426s6.9-19.2 8.4-23.3H486c1.2 5.5 4.8 23.3 4.8 23.3H528z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            class="col-span-7 grid-rows-3"
          >
            <div
              class="flex justify-between text-gray-600"
            >
              <div
                class="text-sm font-bold align-middle"
                data-testid="transaction-name"
              >
                Payment
              </div>
              <div
                class="text-sm font-bold align-text-top"
                data-testid="transaction-amount"
              >
                +$100.00
              </div>
            </div>
            <div
              class="text-sm text-gray-600 elipsis-row align-middle"
              data-testid="transaction-description"
            >
              Test transaction 1
            </div>
            <div
              class="text-sm text-gray-600 elipsis-row align-middle"
              data-testid="transaction-date"
            >
              9/14/2020
            </div>
          </div>
          <div
            class="col-span-1"
          >
            <button
              class="w-full h-full text-gray-400 bg-white hover:bg-gray-200"
            >
              <a
                class="w-full h-full inline-block align-text-bottom"
                data-testid="transaction-detail-button"
                href="/transaction/5f5f3b3b9b1f3b0017f3"
              >
                &gt;
              </a>
            </button>
          </div>
        </div>
        <div
          class="flex-row gap-1 p-1 bg-white grid grid-cols-10"
        >
          <div
            class="col-span-2 gap-1 p-1"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-cc-visa w-full h-full"
              color="black"
              data-icon="cc-visa"
              data-prefix="fab"
              data-testid="transaction-icon"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M470.1 231.3s7.6 37.2 9.3 45H446c3.3-8.9 16-43.5 16-43.5-.2.3 3.3-9.1 5.3-14.9l2.8 13.4zM576 80v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V80c0-26.5 21.5-48 48-48h480c26.5 0 48 21.5 48 48zM152.5 331.2L215.7 176h-42.5l-39.3 106-4.3-21.5-14-71.4c-2.3-9.9-9.4-12.7-18.2-13.1H32.7l-.7 3.1c15.8 4 29.9 9.8 42.2 17.1l35.8 135h42.5zm94.4.2L272.1 176h-40.2l-25.1 155.4h40.1zm139.9-50.8c.2-17.7-10.6-31.2-33.7-42.3-14.1-7.1-22.7-11.9-22.7-19.2.2-6.6 7.3-13.4 23.1-13.4 13.1-.3 22.7 2.8 29.9 5.9l3.6 1.7 5.5-33.6c-7.9-3.1-20.5-6.6-36-6.6-39.7 0-67.6 21.2-67.8 51.4-.3 22.3 20 34.7 35.2 42.2 15.5 7.6 20.8 12.6 20.8 19.3-.2 10.4-12.6 15.2-24.1 15.2-16 0-24.6-2.5-37.7-8.3l-5.3-2.5-5.6 34.9c9.4 4.3 26.8 8.1 44.8 8.3 42.2.1 69.7-20.8 70-53zM528 331.4L495.6 176h-31.1c-9.6 0-16.9 2.8-21 12.9l-59.7 142.5H426s6.9-19.2 8.4-23.3H486c1.2 5.5 4.8 23.3 4.8 23.3H528z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            class="col-span-7 grid-rows-3"
          >
            <div
              class="flex justify-between text-gray-600"
            >
              <div
                class="text-sm font-bold align-middle"
                data-testid="transaction-name"
              >
                Credit
              </div>
              <div
                class="text-sm font-bold align-text-top"
                data-testid="transaction-amount"
              >
                $100.00
              </div>
            </div>
            <div
              class="text-sm text-gray-600 elipsis-row align-middle"
              data-testid="transaction-description"
            >
              Test transaction 2
            </div>
            <div
              class="text-sm text-gray-600 elipsis-row align-middle"
              data-testid="transaction-date"
            >
              9/15/2020
            </div>
          </div>
          <div
            class="col-span-1"
          >
            <button
              class="w-full h-full text-gray-400 bg-white hover:bg-gray-200"
            >
              <a
                class="w-full h-full inline-block align-text-bottom"
                data-testid="transaction-detail-button"
                href="/transaction/5f5f3b3b9b1f3b0017f3"
              >
                &gt;
              </a>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="grid grid-cols-1 gap-0 sm:grid-cols-2 align-text-left font-bold text-gray-800 text-2xl"
      data-testid="transactions-list-header"
    >
      Latest transactions
    </div>
    <div
      class="grid grid-cols-1 gap-0 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 rounded-md"
      data-testid="transactions-list"
    >
      <div
        class="flex-row gap-1 p-1 bg-white grid grid-cols-10"
      >
        <div
          class="col-span-2 gap-1 p-1"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-cc-visa w-full h-full"
            color="black"
            data-icon="cc-visa"
            data-prefix="fab"
            data-testid="transaction-icon"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M470.1 231.3s7.6 37.2 9.3 45H446c3.3-8.9 16-43.5 16-43.5-.2.3 3.3-9.1 5.3-14.9l2.8 13.4zM576 80v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V80c0-26.5 21.5-48 48-48h480c26.5 0 48 21.5 48 48zM152.5 331.2L215.7 176h-42.5l-39.3 106-4.3-21.5-14-71.4c-2.3-9.9-9.4-12.7-18.2-13.1H32.7l-.7 3.1c15.8 4 29.9 9.8 42.2 17.1l35.8 135h42.5zm94.4.2L272.1 176h-40.2l-25.1 155.4h40.1zm139.9-50.8c.2-17.7-10.6-31.2-33.7-42.3-14.1-7.1-22.7-11.9-22.7-19.2.2-6.6 7.3-13.4 23.1-13.4 13.1-.3 22.7 2.8 29.9 5.9l3.6 1.7 5.5-33.6c-7.9-3.1-20.5-6.6-36-6.6-39.7 0-67.6 21.2-67.8 51.4-.3 22.3 20 34.7 35.2 42.2 15.5 7.6 20.8 12.6 20.8 19.3-.2 10.4-12.6 15.2-24.1 15.2-16 0-24.6-2.5-37.7-8.3l-5.3-2.5-5.6 34.9c9.4 4.3 26.8 8.1 44.8 8.3 42.2.1 69.7-20.8 70-53zM528 331.4L495.6 176h-31.1c-9.6 0-16.9 2.8-21 12.9l-59.7 142.5H426s6.9-19.2 8.4-23.3H486c1.2 5.5 4.8 23.3 4.8 23.3H528z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="col-span-7 grid-rows-3"
        >
          <div
            class="flex justify-between text-gray-600"
          >
            <div
              class="text-sm font-bold align-middle"
              data-testid="transaction-name"
            >
              Payment
            </div>
            <div
              class="text-sm font-bold align-text-top"
              data-testid="transaction-amount"
            >
              +$100.00
            </div>
          </div>
          <div
            class="text-sm text-gray-600 elipsis-row align-middle"
            data-testid="transaction-description"
          >
            Test transaction 1
          </div>
          <div
            class="text-sm text-gray-600 elipsis-row align-middle"
            data-testid="transaction-date"
          >
            9/14/2020
          </div>
        </div>
        <div
          class="col-span-1"
        >
          <button
            class="w-full h-full text-gray-400 bg-white hover:bg-gray-200"
          >
            <a
              class="w-full h-full inline-block align-text-bottom"
              data-testid="transaction-detail-button"
              href="/transaction/5f5f3b3b9b1f3b0017f3"
            >
              &gt;
            </a>
          </button>
        </div>
      </div>
      <div
        class="flex-row gap-1 p-1 bg-white grid grid-cols-10"
      >
        <div
          class="col-span-2 gap-1 p-1"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-cc-visa w-full h-full"
            color="black"
            data-icon="cc-visa"
            data-prefix="fab"
            data-testid="transaction-icon"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M470.1 231.3s7.6 37.2 9.3 45H446c3.3-8.9 16-43.5 16-43.5-.2.3 3.3-9.1 5.3-14.9l2.8 13.4zM576 80v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V80c0-26.5 21.5-48 48-48h480c26.5 0 48 21.5 48 48zM152.5 331.2L215.7 176h-42.5l-39.3 106-4.3-21.5-14-71.4c-2.3-9.9-9.4-12.7-18.2-13.1H32.7l-.7 3.1c15.8 4 29.9 9.8 42.2 17.1l35.8 135h42.5zm94.4.2L272.1 176h-40.2l-25.1 155.4h40.1zm139.9-50.8c.2-17.7-10.6-31.2-33.7-42.3-14.1-7.1-22.7-11.9-22.7-19.2.2-6.6 7.3-13.4 23.1-13.4 13.1-.3 22.7 2.8 29.9 5.9l3.6 1.7 5.5-33.6c-7.9-3.1-20.5-6.6-36-6.6-39.7 0-67.6 21.2-67.8 51.4-.3 22.3 20 34.7 35.2 42.2 15.5 7.6 20.8 12.6 20.8 19.3-.2 10.4-12.6 15.2-24.1 15.2-16 0-24.6-2.5-37.7-8.3l-5.3-2.5-5.6 34.9c9.4 4.3 26.8 8.1 44.8 8.3 42.2.1 69.7-20.8 70-53zM528 331.4L495.6 176h-31.1c-9.6 0-16.9 2.8-21 12.9l-59.7 142.5H426s6.9-19.2 8.4-23.3H486c1.2 5.5 4.8 23.3 4.8 23.3H528z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="col-span-7 grid-rows-3"
        >
          <div
            class="flex justify-between text-gray-600"
          >
            <div
              class="text-sm font-bold align-middle"
              data-testid="transaction-name"
            >
              Credit
            </div>
            <div
              class="text-sm font-bold align-text-top"
              data-testid="transaction-amount"
            >
              $100.00
            </div>
          </div>
          <div
            class="text-sm text-gray-600 elipsis-row align-middle"
            data-testid="transaction-description"
          >
            Test transaction 2
          </div>
          <div
            class="text-sm text-gray-600 elipsis-row align-middle"
            data-testid="transaction-date"
          >
            9/15/2020
          </div>
        </div>
        <div
          class="col-span-1"
        >
          <button
            class="w-full h-full text-gray-400 bg-white hover:bg-gray-200"
          >
            <a
              class="w-full h-full inline-block align-text-bottom"
              data-testid="transaction-detail-button"
              href="/transaction/5f5f3b3b9b1f3b0017f3"
            >
              &gt;
            </a>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
